src/algorithm/minkowski/gjk/simplex/mod.rs

```diff
@@ -14,13 +14,25 @@ pub trait SimplexProcessor {
     type Point: EuclideanSpace;
 
     /// Check if the given simplex contains origin, and if not, update the simplex and search
-    /// direction
+    /// direction.
+    ///
+    /// Used by the GJK intersection test
     fn check_origin(
```

Comment:
the function name needs to be changed to express the mutability side of the story